lib/spack/spack/util/web.py

```diff
@@ -292,9 +292,13 @@ def remove_url(url):
 
 
 def _list_s3_objects(client, url, num_entries, start_after=None):
+    url_path = url.path
+    if url_path.startswith('/'):
+        url_path = url_path[1:]
+
```

Comment:
@opadron Any chance you can take a look at this workaround I put in based on our conversation?  It seems to have broken other things.  Maybe the second change of `url.path` for `url_path` was wrong?

Reply:
I guess it would be easiest for you to see it if you isolate that most recent WIP commit though...

Reply:
Ok, so I got to the bottom of it, it wasn't that WIP commit at all, that actually seems to have fixed my pipeline problems.  The real thing that was killing me was that at the same time I pushed the WIP fix we talked about, it also picked up a bunch of commits up to today's `develop`, and one of those broke uploading cdashid files.  I resolved that with my commit `Fix issue that cropped up since #13408 was merged`.  The expectation now is that the local file argument to `push_to_url` is not formatted like a url, but just a path.  So that was an easy fix, once I dug down to the bottom and figured it out ;-)